en/docs/develop/setting-up-user-portal-in-a-dev-environment.md

```diff
@@ -0,0 +1,101 @@
+# Setting-Up User Portal in a Development Environment
+
+In [WSO2 Identity Server 5.10.0](https://wso2.com/identity-and-access-management/) the user portal is available under
+ the URL `http(s)://<SERVER_HOST>/t/<TENANT_DOMAIN>/user-portal`. If you are planning on setting up the repo in a
+ development environment follow the steps given below.
+
+!!! note "Before you begin"
+
+    1. Install [node](https://nodejs.org/en/download/) if you haven’t already(npm is already bundled with node).
+    2. Install [maven](https://maven.apache.org/download.cgi) (Needed to run mvn commands).
+    3. A running Identity Server v 5.10. (If you want to build from source follow the instructions listed [here]
+    (https://github.com/wso2/product-is)).
+
+
+### Step 1: Configuring Identity Server
+
+!!! tip
+    Default dev origin, hostname and port set for the User Portal is https://localhost:9000. So the following
+    configurations will have to change accordingly if you change the port or any other configurations.
+
+
+1.  Allowing Cross Origin requests to the Identity Server.
+
+    Cross origin requests will be blocked by default as a security measure. So you have to put the following CORS
+    filter to the config file `<IS_HOME>/repository/resources/conf/templates/repository/conf/tomcat/web.xml.j2` in
+    WSO2 Identity Server distribution pack.
+    ```
+        <filter>
+            <filter-name>CORS</filter-name>
+            <filter-class>com.thetransactioncompany.cors.CORSFilter</filter-class>
+            <init-param>
+                <param-name>cors.allowOrigin</param-name>
+                <param-value>https://localhost:9000</param-value>
+            </init-param>
+            <init-param>
+                <param-name>cors.supportedMethods</param-name>
+                <param-value>GET, HEAD, POST, DELETE, OPTIONS, PATCH, PUT</param-value>
+            </init-param>
+            <init-param>
+                <param-name>cors.exposedHeaders</param-name>
+                <param-value>Location</param-value>
+            </init-param>
+        </filter>
+        <filter-mapping>
+            <filter-name>CORS</filter-name>
+            <url-pattern>/*</url-pattern>
+            <dispatcher>REQUEST</dispatcher>
+            <dispatcher>FORWARD</dispatcher>
+        </filter-mapping>
+    ```
+2.  Add your hostname and port as a trusted FIDO2 origin.
+
+    Whitelist the dev URL in the FIDO configurations found in the file
```

Comment:
2. Whitelist your hostname and port as a trusted FIDO2 origin.

     Add the dev url to the `<Origin>` tag of the FIDO configurations in the `<IS_HOME>/repository/resources/conf/templates/repository/conf/identity/identity.xml.j2` file.